libnetwork/drivers/overlay/ov_network.go

```diff
@@ -1083,35 +1023,7 @@ func (n *network) obtainVxlanID(s *subnet) error {
 	if n.vxlanID(s) != 0 {
 		return nil
 	}
-
-	if n.driver.store == nil {
-		return fmt.Errorf("no valid vxlan id and no datastore configured, cannot obtain vxlan id")
-	}
-
-	for {
-		if err := n.driver.store.GetObject(datastore.Key(n.Key()...), n); err != nil {
-			return fmt.Errorf("getting network %q from datastore failed %v", n.id, err)
-		}
-
-		if n.vxlanID(s) == 0 {
-			vxlanID, err := n.driver.vxlanIdm.GetID(true)
-			if err != nil {
-				return fmt.Errorf("failed to allocate vxlan id: %v", err)
-			}
-
-			n.setVxlanID(s, uint32(vxlanID))
-			if err := n.writeToStore(); err != nil {
-				n.driver.vxlanIdm.Release(uint64(n.vxlanID(s)))
-				n.setVxlanID(s, 0)
-				if err == datastore.ErrKeyModified {
-					continue
-				}
-				return fmt.Errorf("network %q failed to update data store: %v", n.id, err)
-			}
-			return nil
-		}
-		return nil
-	}
+	return fmt.Errorf("no valid vxlan id and no datastore configured, cannot obtain vxlan id")
```

Comment:
What datastore? Never heard of him. Jokes aside, the error message could use an edit for a global-datastore-less world.

Reply:
Not sure if you're reviewing the whole diff or commits one by one. I didn't change this error message, instead I'm just getting rid of this method in a latter commit (I'm going to push it latter today) :slightly_smiling_face: